### ISSUE TITLE
Discrepancy with Ruby Style Guide regarding regexp and BinaryOperatorParameterName cop

### ISSUE THREAD

**Issue by Zajn** [rubocop-hq/rubocop#8901]
I implemented a class which defined `=~`. I was basing it off a class internal to Rails, and so I also utilized their parameter name of `re`. In addition to being too short, to my surprise, Rubocop informed me that I violated `Naming/BinaryOperatorParameterName` and should name my parameter `other`. I looked at the [Rubocop docs](https://docs.rubocop.org/rubocop/cops_naming.html#namingbinaryoperatorparametername), which linked me to the ruby style guide for the [`other` parameter](https://rubystyle.guide/#other-arg). It is my understanding that Rubocop is based on this guide.

The style guide states:
> When defining binary operators and operator-alike methods, name the parameter other for operators with "symmetrical" semantics of operands. Symmetrical semantics means both sides of the operator are typically of same or coercible types.
>
> Operators and operator-alike methods with symmetrical semantics (the parameter should be named `other`): `+`, `-`, `*`, `/`, `%`, `**`, `==`, `>`, `<`, `|`, `&`, `^`, `eql?`, `equal?`.

the style guide then states
> Operators with non-symmetrical semantics (the parameter should not be named `other`): `<<`, `[]` (collection/item relations between operands), `===` (pattern/matchable relations). 

I would assume that `=~` falls under pattern/matchable relations, but I might be wrong. I would expect in this case, that I would not be in violation of the `Naming/BinaryOperatorParameterName` cop, especially considering `=~` is not listed in the operators and operator-alike methods.

For example's sake, here is the aforementioned method that was in violation:

```ruby
def =~(re)
  re.match?(to_s)
end
```
--------

## Expected behavior
I expected to not be in violation of `Naming/BinaryOperatorParameterName` for this method.

## Actual behavior

Describe here what actually happened.
Please use `rubocop --debug` when pasting rubocop output as it contains additional information.

```ruby
For /Users/zach/workspace/checklistpro: configuration from /Users/zach/workspace/checklistpro/.rubocop.yml
configuration from /Users/zach/.gem/ruby/2.7.2/gems/rubocop-performance-1.8.1/config/default.yml
configuration from /Users/zach/.gem/ruby/2.7.2/gems/rubocop-performance-1.8.1/config/default.yml
Default configuration from /Users/zach/.gem/ruby/2.7.2/gems/rubocop-0.93.1/config/default.yml
configuration from /Users/zach/.gem/ruby/2.7.2/gems/rubocop-rails-2.8.1/config/default.yml
configuration from /Users/zach/.gem/ruby/2.7.2/gems/rubocop-rails-2.8.1/config/default.yml
Inheriting configuration from /Users/zach/workspace/checklistpro/.rubocop_todo.yml
Inspecting 1 file
Scanning /Users/zach/workspace/checklistpro/app/models/jetpack_time_zone.rb
C

Offenses:

app/models/jetpack_time_zone.rb:28:10: C: Naming/BinaryOperatorParameterName: When defining the =~ operator, name its argument other.
  def =~(re)
         ^^
app/models/jetpack_time_zone.rb:28:10: C: Naming/MethodParameterName: Method parameter must be at least 3 characters long.
  def =~(re)
         ^^

1 file inspected, 2 offenses detected
Finished in 0.5917779998853803 seconds
```

## Steps to reproduce the problem
Run the following code with `ruby hello.rb` to verify it works. Then run `rubocop hello.rb` and verify that the `def =~` method is in violation of `Naming/BinaryOperatorParameterName`.

![image](https://user-images.githubusercontent.com/1636886/96286624-ac0a1480-0fae-11eb-9746-5cb7c2758505.png)

```ruby
# hello.rb
# frozen_string_literal: true

# Example of Naming/BinaryOperatorParameterName cop
class Hello
  def initialize(name)
    @name = name
  end

  def =~(regex)
    regex.match?('Zach')
  end

  def self.greet(name)
    h = new(name)

    puts 'Hello Zach' if h =~ /#{name}/
  end
end

Hello.greet('Fred')
Hello.greet('Zach')
```
## RuboCop version
```
~/workspace/rubocop-example » rubocop -V                                                                                                
0.93.1 (using Parser 2.7.2.0, rubocop-ast 0.8.0, running on ruby 2.7.2 x86_64-darwin19)
```

------

I would be happy to submit a PR if this is something that should be corrected. If not, thanks for your time!

**Comment by marcandre** [comment]
Very good research and reasoning 💪!

I agree this is a false positive for `BinaryOperatorParameterName`; PR welcome 😄

**Comment by Zajn** [comment]
> I agree this is a false positive for `BinaryOperatorParameterName`; PRs welcome 😄

Ok, great! I'd be happy to put one together.
